### PR TITLE
Replace tsc/babel with swc

### PR DESCRIPTION
**Pull request by birkskyum**
Our jest testing, and transpilation in general is performed with Babel and tsc. This PR replaces them with SWC for a significantly speed up to that part of our toolchain.

On my machine:
- Tanspilation of src and bench folder: 14 s => 3 s,
- Unit tests: ~60 s => ~12 s

The reason it's faster it to be found in the underlying architechture where SWC is written in Rust with more degrees of freedom, and TSC/Babel are js/ts projects

A bit of background on this topic
https://blog.logrocket.com/why-you-should-use-swc/
https://engineering.telia.no/blog/we-exchanged-babel-tsc-for-swc-and-it-was-worth-it

### CODE REVIEW

File: jest.config.cjs [hunk 1]
```diff
@@ -9,8 +9,7 @@ module.exports = {
     "**/?(*.)+(spec|test).+(ts|tsx|js)"
   ],
   "transform": {
-    "^.+\\.(ts|tsx)$": "ts-jest",
-    "^.+\\.(js)$": "babel-jest",
+    '^.+\\.(t|j)sx?$': '@swc/jest',
     "^.+\\.(glsl)$": "jest-raw-loader",
```
[the remark this answers is not in the document]
Comment (birkskyum) [reply jest.config.cjs:13]: Not anymore actually

[the remark this answers is not in the document]
Comment (birkskyum) [reply jest.config.cjs:13]: Good catch. Made a separate PR in here #937

[the remark this answers is not in the document]
Comment (HarelM) [reply jest.config.cjs:13]: Thanks!

**Comment by HarelM** [comment]
Npm trends shows that swc is far from being as popular as babel.
I tend to wait until it gains more popularity as I fear there will be issues related to typescript due to its immaturity...

**Comment by birkskyum** [comment]
It's behind the scenes the foundation of parcel 2 and next 12, so with that in mind, the adoption is actually quite fast https://www.npmtrends.com/@swc/core-vs-next-vs-parcel

**Comment by HarelM** [comment]
@nyurik you seemed excited about this in the slack channel. Feel free to weigh in here... :-)
I think this is a small change and easy to revert, but I would hate if someone would waste a lot of time trying to understand why something is not working as expected and find out that the cause is this tool and that we adopted it too early.

**Comment by nyurik** [comment]
@HarelM you do have a point, and it is a bit tricky to test these kinds of changes. I would recommend releasing this change as its own minor (not patch) change, and if possible, wait for a bit between releasing last version and the next (and don't merge anything else in that time window).

**Comment by wipfli** [comment]
The motivation for this pull request seems to be faster compilation, i.e., better developer experience. However, if we cannot merge anything to main for a while, this would have a negative impact on how we work.

**Comment by HarelM** [comment]
I agree, not being able to merge for a while is not a good option.
Let's try it out, we could always revert back if we feel some things are not working as expected.
Currently, this only affects the tests which for me is an enabler to running the tests locally.
We should remember though, add a development team, that we should be suspicious of this change and flag things that we see that are causing issues, if any.

**Comment by birkskyum** [comment]
Thanks for all the thoughts. I sense a consensus on carefully moving forward with this, and allowing other merges as well, while being extra attentive to issues that in any way could be related to this change.

**Review by HarelM — APPROVED** [review]

**Comment by wipfli** [comment]
Sounds good to me. Is the `eval()` really needed?

**Comment by birkskyum** [comment]
It seemed to me that try {eval} catch{} was a simpler more direct way to test "evaluates without errors" without having to use async/await, dynamic imports etc.

**Comment by wipfli** [comment]
If the change to `eval` was not needed for swc, then it would have been better placed in a separate pull request. But never mind, I was just curious...

**Comment by wipfli** [comment]
Sorry for nagging, but what would hang at this step? The CI of this pull request or the CI of main before this was merged?

**Comment by birkskyum** [comment]
The test-build of this pull request, because the dynamic import within try/catch was hanging.

**Comment by wipfli** [comment]
Did you look into https://github.com/swc-project/swc/issues/140?

**Comment by birkskyum** [comment]
I didn't investigate further, because dynamic import is a feature of es2020, async/await is from es2017, and we ship es2015 at this point, in umd/amd/iife/commonjs formats, so the transpilation and modularization is quite convoluted right now.

I would love for us to simplify things with a newer language standard (see #881), and maybe even esm throughout, but there are still a few blockers holding us back, such as [web workers in firefox](https://caniuse.com/mdn-api_worker_worker_ecmascript_modules), and until, I believe we might be better off avoiding too new features as they just add polyfills.

**Pull request merged**